crates/libgraphql-core/src/test/golden_tests/test_runner.rs

```diff
@@ -0,0 +1,632 @@
+            snippet.push_str(&format!("   {line_num:>3} → {line} ⚠️ (error not raised!)\n"));
+        } else if idx < 5 {
+            snippet.push_str(&format!("   {line_num:>3} │ {line}\n"));
+        }
+    }
+
+    Ok(snippet)
+}
+
+/// Run all operation validation golden tests
+pub fn run_operation_tests(fixtures_dir: &Path) -> GoldenTestResults {
+    let test_cases = GoldenTestCase::discover_all(fixtures_dir);
+
+    #[cfg(test)]
+    let test_results: Vec<GoldenTestResult> = test_cases
+        .par_iter()
+        .filter(|test_case| test_case.schema_should_be_valid)
+        .filter_map(|test_case| {
+            // Build the schema first
+            let schema = try_build_schema(&test_case.schema_paths)?;
+
+            // Test valid operations
+            let mut results = test_valid_operations(test_case, &schema);
+
+            // Test invalid operations
+            let invalid_results = test_invalid_operations(test_case, &schema);
+            results.extend(invalid_results);
+
+            Some(results)
+        })
+        .flatten()
+        .collect();
+
+    #[cfg(not(test))]
+    let test_results: Vec<GoldenTestResult> = test_cases
+        .iter()
+        .filter(|test_case| test_case.schema_should_be_valid)
+        .filter_map(|test_case| {
+            // Build the schema first
+            let schema = try_build_schema(&test_case.schema_paths)?;
+
+            // Test valid operations
+            let mut results = test_valid_operations(test_case, &schema);
+
+            // Test invalid operations
+            let invalid_results = test_invalid_operations(test_case, &schema);
+            results.extend(invalid_results);
+
+            Some(results)
+        })
+        .flatten()
+        .collect();
+
+    let mut results = GoldenTestResults::new();
+    results.extend(test_results);
+    results
+}
+
+/// Helper to try building a schema from multiple files
+fn try_build_schema(schema_paths: &[PathBuf]) -> Option<Schema> {
+    let mut builder = SchemaBuilder::new();
+
+    for schema_path in schema_paths {
+        builder = match builder.load_file(schema_path) {
+            Ok(b) => b,
+            Err(_) => return None,
+        };
+    }
+
+    builder.build().ok()
+}
+
+/// Test valid operations against a schema
+fn test_valid_operations(test_case: &GoldenTestCase, schema: &Schema) -> Vec<GoldenTestResult> {
+    let mut results = Vec::new();
+
+    if test_case.valid_operations.is_empty() {
+        return results;
+    }
+
+    // Test each valid operation with its own fragment registry
+    for op_test in &test_case.valid_operations {
+        let test_name = format!(
+            "{}/valid_operations/{}",
+            test_case.name,
+            op_test.path.file_name().unwrap().to_str().unwrap()
+        );
+
+        // Build fragment registry from just this operation file
+        let fragment_registry = match build_fragment_registry(schema, &[&op_test.path]) {
```

Comment:
I think we want to build one `FragmentRegistry` for the whole test suite, then re-use that `FragmentRegistry` for each operation...

If two operation test_cases define fragments with the same name, we need to update those conflicting fragment names to be unique within the test suite.